aws-datastore/src/main/java/com/amplifyframework/datastore/model/SimpleModelProvider.java

```diff
@@ -146,22 +146,24 @@ public String toString() {
     /**
      * Configures and builds instances of SimpleModelProvider.
      */
-    static final class Builder {
+    @SuppressWarnings("WeakerAccess")
+    public static final class Builder {
```

Comment:
Won't the public methods require javadoc?

Reply:
Should our checkstyle work this way? Yes, I think so.

Does it actually? Unfortunately not. When I built the rule-set last year, I forgot to add [`MissingJavadocMethod`](https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocMethod).

Later, we added [`MissingJavadocType`](https://github.com/aws-amplify/amplify-android/blob/master/configuration/checkstyle-rules.xml#L245), but we still don't have the method-level presence check.

We _do_ have checks that say like "_If_ it is there, it had dang-nabbit better be well formed as heck!".

As a result, I think we probably have a number of places in the code that are missing Javadocs. Anyway, I will add them for _these_, since we observed the issue in code review.

As a separate exercise, we should go back and add the `MissingJavadocMethod` for `protected` and `public`s, so that we can catch it before code review, in the future!

Reply:
Ehh, actually update -- I'll just add them all when I add `MissingJavadocType`. Cause that likely will require adding several more, anyway.